src/org/loklak/api/search/GithubProfileScraper.java

```diff
@@ -0,0 +1,214 @@
+/**
+ *  Github Profile Crawler
+ *  Copyright 22.07.2016 by Jigyasa Grover, @jig08
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *  
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *  
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this program in the file lgpl21.txt
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.loklak.api.search;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import javax.servlet.http.HttpServletResponse;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.loklak.server.APIException;
+import org.loklak.server.APIHandler;
+import org.loklak.server.AbstractAPIHandler;
+import org.loklak.server.Authorization;
+import org.loklak.server.BaseUserRole;
+import org.loklak.server.Query;
+import org.loklak.susi.SusiThought;
+import org.loklak.tools.storage.JSONObjectWithDefault;
+
+public abstract class GithubProfileScraper extends AbstractAPIHandler implements APIHandler {
+
+	private static final long serialVersionUID = -4166800345379685201L;
+
+	@Override
+	public String getAPIPath() {
+		return "/api/githubprofilescraper.json";
+	}
+
+	@Override
+	public BaseUserRole getMinimalBaseUserRole() {
+		return BaseUserRole.ANONYMOUS;
+	}
+
+	@Override
+	public JSONObject getDefaultPermissions(BaseUserRole baseUserRole) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	public JSONObject serviceImpl(Query call, HttpServletResponse response, Authorization rights, JSONObjectWithDefault permissions)
+			throws APIException {
+		String profile = call.get("profile", "");
+		return scrapeGithub(profile);
+	}
+
+	public static SusiThought scrapeGithub(String profile) {
+
+		Document html = null;
+
+		JSONObject githubProfile = new JSONObject();
+
+		try {
+			html = Jsoup.connect("https://github.com/" + profile).get();
+		} catch (IOException e) {
+			
+			URI uri = null;
+			try {
+				uri = new URI("https://api.github.com/search/users?q=" + profile);
+			} catch (URISyntaxException e1) {
+				e1.printStackTrace();
+			}
+			
+			JSONTokener tokener = null;
+			try {
+				tokener = new JSONTokener(uri.toURL().openStream());
```

Comment:
>Is this step necessary can't we have URI defined in the same line as it is used ? since this uri variable isn't being used outside this catch.
URI uri = new URI() should be possible right ?

@sudheesh001 Probably not. 
`uri` is being used here for `JSONTokener`